### PR TITLE
Add module hana_query to make SAP HANA administration easier.

### PR DESCRIPTION
**Pull request by rainerleber**
SUMMARY
Add module hana_query for SAP HANA administration - SQL querys.
This module will call SQL Statements from files or as string against a SAP HANA DB

ISSUE TYPE
New Module Pull Request
COMPONENT NAME
hana_query

### CODE REVIEW

**Comment by rainerleber** [comment]
I have done a lot of mistakes, so i reopend it.

**Comment by rainerleber** [comment]
@Andersson007 I do a lot of mistakes cloud you please approve this pr again ?

**Comment by rainerleber** [comment]
Thank you what are the next steps ? :-)

**Comment by Andersson007** [comment]
feedback from @felixfontein :)

File: tests/unit/plugins/modules/database/saphana/test_hana_query.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            'instance': "01",
+            'encrypted': False,
+            'host': "localhost",
+            'user': "SYSTEM",
+            'password': "1234Qwer",
+            'database': "HDB",
+            'query': "SELECT * FROM users;"
+        })
+        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
+            run_command.return_value = 0, '', ''  # successful execution, no output
```
Comment (felixfontein) [inline tests/unit/plugins/modules/database/saphana/test_hana_query.py:59]: As mentioned in the previous PR, you should add some interesting return values to stdout, and check whether the module parses and returns them correctly in the assertions below.

File: plugins/modules/database/saphana/hana_query.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    if query is not None:
+        for q in query:
+            # makes a command like hdbsql -i 01 -u SYSTEM -p secret123# "select user_name from users",
+            # iterates through multiple commands and append the output to var out.
+            query_command = command + [q]
+            (rc, out_raw, err) = module.run_command(query_command)
+            out.append(csv_to_list(out_raw))
+    changed = True
+
+    module.exit_json(changed=changed, message=rc, query_result=out, stderr=err)
```
Comment (felixfontein) [inline plugins/modules/database/saphana/hana_query.py:183]:
```suggestion
    module.exit_json(changed=changed, rc=rc, query_result=out, stderr=err)
```
I would expect a string for `message`, not an integer.

File: plugins/modules/database/saphana/hana_query.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+# Copyright: (c) 2021, Rainer Leber <rainerleber@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: hana_query
+short_description: Execute SQL on HANA
```
Comment (felixfontein) [inline plugins/modules/database/saphana/hana_query.py:11]:
```suggestion
short_description: Execute SQL on HANA
version_added: 3.2.0
```

File: tests/unit/plugins/modules/database/saphana/test_hana_query.py (hunk 1 above)
Comment (rainerleber) [reply tests/unit/plugins/modules/database/saphana/test_hana_query.py]: Done :-D

File: tests/unit/plugins/modules/database/saphana/test_hana_query.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            'instance': "01",
+            'encrypted': False,
+            'host': "localhost",
+            'user': "SYSTEM",
+            'password': "1234Qwer",
+            'database': "HDB",
+            'query': "SELECT * FROM users;"
+        })
+        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
+            run_command.return_value = 0, 'username\ntestuser\n', ''
```
Comment (felixfontein) [inline tests/unit/plugins/modules/database/saphana/test_hana_query.py:59]: Can you extend this example minimally so it contains more than one column, and also has some spaces that need to be trimmed?

Comment (rainerleber) [reply tests/unit/plugins/modules/database/saphana/test_hana_query.py]: sure, I extended it :-)

File: tests/unit/plugins/modules/database/saphana/test_hana_query.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            'user': "SYSTEM",
+            'password': "1234Qwer",
+            'database': "HDB",
+            'query': "SELECT * FROM users;"
+        })
+        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
+            run_command.return_value = 0, 'username,name\n  testuser,test user  \n myuser, my user   \n', ''
+            with self.assertRaises(AnsibleExitJson) as result:
+                hana_query.main()
+            self.assertEqual(result.exception.args[0]['query_result'], [[{'username': 'testuser', 'name': 'test user'},{'username': 'myuser', 'name': 'my user',}]])
```
Comment (felixfontein) [inline tests/unit/plugins/modules/database/saphana/test_hana_query.py:62]: That should also take care of the sanity test which will fail.
```suggestion
            self.assertEqual(result.exception.args[0]['query_result'], [[
                {'username': 'testuser', 'name': 'test user'},
                {'username': 'myuser', 'name': 'my user'},
            ]])
```

**Review by felixfontein — APPROVED** [review]

**Comment by Andersson007** [comment]
https://github.com/ansible-collections/community.general/pull/2647

**Comment by felixfontein** [comment]
@rainerleber thanks for contributing this!
@Andersson007 thanks for reviewing!

**Pull request merged**